packages/playground/blueprints/src/lib/steps/define-site-url.ts

```diff
@@ -0,0 +1,21 @@
+import { UniversalPHP } from '@php-wasm/universal';
+import { patchFile } from './common';
+
+/**
+ * Sets site URL of the WordPress installation.
+ *
+ * @param playground The playground client.
+ * @param siteUrl
+ */
+export async function defineSiteUrl(playground: UniversalPHP, siteUrl: string) {
+	await patchFile(
+		playground,
+		`/wordpress/wp-config.php`,
+		// @TODO: Remove any existing define('WP_HOME') and define('WP_SITEURL') lines.
```

Comment:
I guess the idea is that someone might supply a `wp-config` with these values defined?

the default WP install _doesn't_, so I think it's fine, but also not necessary 🤷‍♂️ 

one easy hack to do this would be the following best-effort solution…

```js
contents => {
	const withoutExistingSiteUrl = contents.replace(
		/^(\s*define\(\s*)(['"])(WP_(?:HOME|SITEURL))\2(\s*),/g,
		'$1$2X_$3$2$4,'
	);
}
```

…this is a decent way to avoid having to parse the PHP there, as it cannot stand on its own apart from inside a HEREDOC/NOWDOC string, in which case I think the transform would be benign anyway

![Uploading Screenshot 2023-04-24 at 3.17.54 PM.png…]()

Reply:
aha, in retrospec it looks like this has an interplay with `pathSiteUrl()` - is that why we're removing the `define()`?

Reply:
I was worried about this step being ran twice, perhaps as a result of merging multiple blueprints, which would yield warnings about constant WP_HOME being already defined.

Now that I think about it, the code snippet could be wrapped in `if(!defined('WP_HOME')) { }` to remove that risk – then we wouldn't have to modify previously written code at all. Alternatively, there could be an entirely new step called `predefineConstants({})` which would create or update a file like `constants.json`. It would also add a snippet of PHP code to `wp-config.php` to process that file, but it wouldn't add the snippet if it was already added before. In this scenario, `defineSiteUrl()` would be just a wrapper over `predefineConstants()`

Reply:
Yeah "last one wins" sounds like the only way to do it.